### ISSUE TITLE
cabal-install 1.18.0.3 installation fails because of missing `network-uri` build dependency

### ISSUE THREAD

**Issue by geraldus** [NixOS/nixpkgs#6081]
I'm on NixOS 14.12
```bash
Building cabal-install-1.18.0.3...
Preprocessing executable 'cabal' for cabal-install-1.18.0.3...

Distribution/Client/Types.hs:31:8:
    Could not find module ‘Network.URI’
    It is a member of the hidden package ‘network-uri-2.6.0.1’.
    Perhaps you need to add ‘network-uri’ to the build-depends in your .cabal file.
    Use -v to see a list of the files searched for.
```
I suppose this could be easily fixed [[1]](https://github.com/NixOS/nixpkgs/blob/release-14.12/pkgs/tools/package-management/cabal-install/1.18.0.3.nix#L15):
```
buildDepends = [
  Cabal filepath HTTP mtl network network-uri random stm time zlib
];
```
I use version 1.18, because `ghc-mod` does not work with cabal 1.22 and GHC 7.8.3. I've noticed that there is also cabal-install ver.1.20, I'll try it now _(hope ghc-mod will work!)_.

**Comment by geraldus** [comment]
version 1.20.0.3 installed successfully

**Comment by peti** [comment]
Fixed in 4043e6b2d62a3d2a4a5970f76323dfc5bf9ddde4.
